Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,723 @@
+            await services[0].DeleteAsync();
+        }
+
+        [TestMethod]
+        //Start with four workers and four partitions. Then kill one worker.
+        //Test that the partitions will be rebalanced among he three left workers.
+        public async Task TestKillOneWorker()
+        {
+            var services = new AzureStorageOrchestrationService[4];
+            string testName = nameof(TestKillOneWorker);
+
+            for (int i = 0; i < 4; i++)
+            {
+                var settings = new AzureStorageOrchestrationServiceSettings
+                {
+                    TaskHubName = testName,
+                    StorageConnectionString = this.connection,
+                    AppName = testName,
+                    UseTablePartitionManagement = true,
+                    WorkerId = i.ToString(),
+                };
+                services[i] = new AzureStorageOrchestrationService(settings);
+            }
+
+            var startTasks = services.Select(service => service.StartAsync());
+            await Task.WhenAll(startTasks);
+
+            // wait for the partitions to be distributed equally among four workers.
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(30),
+                condition: () =>
+                {
+                    var partitions = services[0].ListTableLeases();
+                    Assert.AreEqual(4, partitions.Count());
+                    return ((partitions.Count(p => p.CurrentOwner == "0") == 1) &&
+                                (partitions.Count(p => p.CurrentOwner == "1") == 1) &&
+                                (partitions.Count(p => p.CurrentOwner == "2") == 1) &&
+                                (partitions.Count(p => p.CurrentOwner == "3") == 1));
+                });
+
+            services[3].KillPartitionManagerLoop();
+
+            bool isBalanced = false;
+
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(40),
+                condition: () =>
+                {
+                    var partitions = services[0].ListTableLeases();
+                    Assert.AreEqual(4, partitions.Count());
+
+                    // Assert that two partitions have the same CurrentOwner value and the other two have unique CurrentOwner values
+                    int distinctCurrentOwnersCount = partitions.Select(x => x.CurrentOwner).Distinct().Count();
+                    isBalanced = distinctCurrentOwnersCount == 3;
+                    return isBalanced;
+                });
+
+            IList<Task> tasks = new List<Task>();
+            tasks.Add(services[0].StopAsync());
+            tasks.Add(services[1].StopAsync());
+            tasks.Add(services[2].StopAsync());
+            await Task.WhenAll(tasks);
+            await services[0].DeleteAsync();
+        }
+
+        //Start with four workers and four partitions. Then kill three workers.
+        //Test that the left worker will take all the partitions.
```

Comment:
nit: let's add a space between // and the first character of each line
Also a minor phrasing suggestion
```suggestion
        // Start with four workers and four partitions. Then kill three workers.
        // Test that the remaining worker will take all the partitions.
```